### ISSUE TITLE
Client Language unavailable before Data Model initialization

### ISSUE THREAD

**Issue by esb** [foundryvtt/foundryvtt#11911]
### What happened?

The data model provides a ``choices`` option that can be used populate ``formGroup`` select options.

The ``choices`` lists need to be made available in the ``Init`` hook.

However, it appears that the data model initiallization occurs BEFORE ``Localization`` is initialized or even before ``game.settings`` are available, which means that the client language is not known. This in turn means that it is impossible to provide a choices list that is sorted using the client's chosen language.

### What ways of accessing Foundry can you encounter this issue in?

- [ ] Native App (Electron)
- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Other

### Reproduction Steps

Insert a debug statement during ``Init`` to print ``game.i18n.language`` and set a non-English client language. The client language will not be available until ``i18nInit``.

### What core version are you reporting this for?

Version 12 Build 331

### Bug Checklist

- [X] The issue occurs while all Modules are disabled

**Comment by esb** [comment]
One way out of this is to provide an optional ``sort`` option on the ``formGroup`` helper that will sort the ``choices`` list at localization time on the data model.

**Comment by stwlam** [comment]
While I'm not clear on what is meant by "data model initialization" since that occurs whenever a model instance is constructed, a field's `choices` can be a function if their enumeration needs to be deferred for any reason. Does that help?

**Comment by esb** [comment]
Yes, I am aware that a function can be used, but that does mean constantly sorting the same static data.

If I try to use a static list of choices, they need to be defined in  ``Init`` otherwise they are ignored. This refers to a data model field in an Item document.

On a related issue, it appears that the only options for a StringDataField ``choices`` option are objects, arrays of strings or functions. This restriction actually makes sorting harder than it needs to be. It would be heaps easier if an array of 2 element arrays were also accepted as they can be easily sorted without having to convert them to and from objects.
